### DISCUSSION TITLE
Performing mutations on returned record(s). Is it possible?

### DISCUSSION THREAD

**Discussion by adxmcollins** [prisma/prisma#5580] [Q&A] [answered]
I've read over the docs and I'm guessing this isn't possible with Prisma, but I thought I'd just ask.

Is something like the below possible?

Where we already have our result from the database saved as a variable and we can perform a mutation on it:

```js
const user = await prisma.users.findFirst({
  where: {
    id: 1
  }
})

user.update({
  data: {
    active: true
  }
})
```

I don't have a specific use-case at the moment, but I'm looking into Prisma to possibly use it in future and it's just something that I noticed when looking over the docs.

**Comment by ruheni** [comment, marked as answer]
Hi @adxmcollins  👋🏾

If I understand correctly, you'd like to perform an update on the user.

You can update a user record using the [`update` ](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#update) in one query.

Your database query will look something close to this: 
```js
const user = prisma.user.update({
  where: {
    id: 1
  },
  data: {
    active: true
  }
})
```

You can learn more about Why Prisma [here](https://www.prisma.io/docs/concepts/overview/why-prisma#tldr). Feel free to let us know if you run into any challenges 🙂

**Reply by adxmcollins** [reply]
Hey, thanks for coming back to me @ruheni !

I'm aware of the `update` function, I was just wondering if there's any way to update a user record we already have saved into a variable (from a previous `find`) - as in the example provided.

I am guessing not as the docs don't mention anything like that, I was just curious if it was possible.

Thanks anyway! 😃

**Reply by ruheni** [reply]
Ah! Thank you for the clarification. It seems like you're looking for something similar to the ActiveRecord pattern. Prisma indeed doesn't support this, there are no model instances but all queries are directly sent via your `PrismaClient` instance. In that sense, Prisma Client acts more like a query builder rather than a traditional ORM.

Hope that helps, let me know if you have any further questions. Always happy to help 🙌🏾

**Reply by adxmcollins** [reply]
Yeah that's it, I couldn't think of the name of what I was trying to explain!

Thanks for the explanation!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
